gms/gossiper.cc

```diff
@@ -798,7 +798,8 @@ future<> gossiper::do_status_check() {
             if (!host_id) {
                 on_internal_error_noexcept(logger, format("Endpoint {} is dead and expired, but unexpecteduly, it has no HOST_ID in endpoint state", endpoint));
             }
-            if (!host_id || !get_token_metadata_ptr()->is_normal_token_owner(host_id)) {
+            const auto* node = get_token_metadata_ptr()->get_topology().find_node(host_id);
+            if (!host_id || !node || !node->is_member()) {
```

Comment:
`do_status_check()` runs only in gossiper mode which does not even support zero token nodes, but is this change noop for the gossiper mode? Does not seams like it to me. Before the change a node in "left" state would have been removed since it is not a normal token owner, but after the change it will not be.

Reply:
This change should be noop for the gossip mode. Leaving nodes are token owners.

As a remainder, this change is necessary to make zero-token nodes work in the recovery mode.